src/crimson/common/interruptible_future.h

```diff
@@ -420,13 +427,23 @@ class [[nodiscard]] interruptible_future_detail<InterruptCond, seastar::future<T
 	(void*)interrupt_cond<InterruptCond>.interrupt_cond.get(),
 	typeid(InterruptCond).name());
       interrupt_cond<InterruptCond>.reset();
-      auto&& value = core_type::get();
-      interrupt_cond<InterruptCond>.set(interruption_condition);
-      INTR_FUT_DEBUG(
-	"interruptible_future_detail::get() got, interrupt_cond: {},{}",
-	(void*)interrupt_cond<InterruptCond>.interrupt_cond.get(),
-	typeid(InterruptCond).name());
-      return std::move(value);
+      try {
+	auto&& value = core_type::get();
+	interrupt_cond<InterruptCond>.set(interruption_condition);
+	INTR_FUT_DEBUG(
+	  "interruptible_future_detail::get() got, interrupt_cond: {},{}",
+	  (void*)interrupt_cond<InterruptCond>.interrupt_cond.get(),
+	  typeid(InterruptCond).name());
+	return std::move(value);
+      } catch (std::exception &e) {
+	interrupt_cond<InterruptCond>.set(interruption_condition);
+	INTR_FUT_DEBUG(
+	  "interruptible_future_detail::get() error {}, interrupt_cond: {},{}",
+	  e,
+	  (void*)interrupt_cond<InterruptCond>.interrupt_cond.get(),
+	  typeid(InterruptCond).name());
+	throw e;
```

Comment:
Hmm, this is basically rethrowing which brings some extra concerns (throwing orignal vs new object). https://stackoverflow.com/a/47618802 might be useful.